packages/application-config/src/process-config.ts

```diff
@@ -28,21 +40,182 @@ type ProcessConfigOptions = Partial<LoadingConfigOptions> & {
+    ].join(''),
+    initialProjectKey:
+      // For the `account` application, we should unset the projectKey.
+      entryPointUriPath === 'account'
+        ? undefined
+        : appConfig.env.development.initialProjectKey,
+    ...(appConfig.env.development?.teamId && {
+      teamId: appConfig.env.development.teamId,
+      // When using a teamId we need to send the actual application of view identifier.
+      ...(isCustomViewData(configurationData)
+        ? { customViewId: configurationData.id }
+        : { applicationId: configurationData.id }),
+    }),
+    oAuthScopes: appConfig.oAuthScopes,
+    additionalOAuthScopes: appConfig?.additionalOAuthScopes,
+  });
+
+  if (isCustomViewData(configurationData)) {
+    const hostUriPath = (appConfig as JSONSchemaForCustomViewConfigurationFiles)
+      .env.development.hostUriPath;
+    const defaultHostUriPath = oidcConfig.initialProjectKey
+      ? `/${oidcConfig.initialProjectKey}/${entryPointUriPath}`
+      : `/${entryPointUriPath}`;
+    const hostUrl = new URL(
+      hostUriPath || defaultHostUriPath,
+      'http://localhost:3001'
+    );
+    return omitDevConfigIfEmpty<ApplicationRuntimeEnvironmentForDevelopment>({
+      oidc: oidcConfig,
+      customViewConfig: configurationData,
+      customViewHostUrl: hostUrl.href,
+    });
+  }
+
+  return omitDevConfigIfEmpty<ApplicationRuntimeEnvironmentForDevelopment>({
+    oidc: oidcConfig,
+    menuLinks: {
+      icon: configurationData.icon,
+      ...configurationData.mainMenuLink,
+      submenuLinks: configurationData.submenuLinks,
+    },
+    // @ts-expect-error: the `accountLinks` is not explicitly typed as it's only used by the account app.
+    accountLinks: configurationData.accountLinks,
+  });
+};
+
+const getRuntimeEnvironmentConfig = ({
+  isProd,
+  configurationData,
+  additionalAppEnv,
+  mcApiUrl,
+  cdnUrl,
+  appUrl,
+  appEnvKey,
+  revision,
+  appConfig,
+}: {
+  isProd: boolean;
+  configurationData: CustomApplicationData | CustomViewData;
+  additionalAppEnv: Record<string, unknown>;
+  mcApiUrl: URL;
+  cdnUrl: URL;
+  appUrl: URL;
+  appEnvKey: string;
+  revision: string;
+  appConfig:
+    | JSONSchemaForCustomApplicationConfigurationFiles
+    | JSONSchemaForCustomViewConfigurationFiles;
+}): ApplicationRuntimeEnvironment => {
+  const entryPointUriPath = isCustomViewData(configurationData)
+    ? // When the application acts as the host for Custom Views, there is no real
+      // entry point to be used, therefore we use a special identifier.
+      CUSTOM_VIEW_HOST_ENTRY_POINT_URI_PATH
+    : configurationData.entryPointUriPath;
+
+  // The real application ID is only used in production.
+  // In development, we prefix the entry point with the "__local" prefix.
+  // This is important to determine to which URL the MC should redirect to
+  // after successful login.
+  const applicationIdentifier = isProd
+    ? `${configurationData.id}:${entryPointUriPath}`
+    : `__local:${entryPointUriPath}`;
```

Comment:
Then we have the `applicationIdentifier`, which is what we send as `X-Application-Id` header.

After some back and forth, I realized this is a required field that can be use for both scenarios: Custom Applications and Custom Views.

As mentioned above, even for Custom Views we still need a `entryPointUriPath` even though it's a special value. Therefore, we can keep the same format for the `applicationIdentifier` of Custom Applications.

Examples:

```
# Custom Applications
123:avengers
__local:avengers

# Custom Views
123:@@custom-view-host@@
__local:@@custom-view-host@@
```